engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -348,34 +361,32 @@ class WorkflowManagerActor(config: Config)
   /** Submit the workflow and return an updated copy of the state data reflecting the addition of a
     * Workflow ID -> WorkflowActorRef entry.
     */
-  private def submitWorkflow(source: WorkflowSourceFiles, replyTo: Option[ActorRef], id: Option[WorkflowId]): WorkflowManagerData = {
+  private def submitWorkflow(source: WorkflowSourceFiles, replyTo: Option[ActorRef], id: Option[WorkflowId]): Future[(WorkflowId, ActorRef)] = {
     val workflowId: WorkflowId = id.getOrElse(WorkflowId.randomId())
     logger.info(s"$tag submitWorkflow input id = $id, effective id = $workflowId")
     val isRestart = id.isDefined
 
-    case class ActorAndStateData(actor: WorkflowActorRef, data: WorkflowManagerData)
-
-    val actorAndData = for {
-      descriptor <- Try(WorkflowDescriptor(workflowId, source, config))
-      actor = context.actorOf(WorkflowActor.props(descriptor), s"WorkflowActor-$workflowId")
-      _ = actor ! SubscribeTransitionCallBack(self)
-      data = stateData.add(workflowId -> actor)
-    } yield ActorAndStateData(actor, data)
-
-    actorAndData match {
-      case Failure(e) =>
-        val messageOrBlank = Option(e.getMessage).mkString
-        logger.error(e, s"$tag: Workflow failed submission: " + messageOrBlank)
-        replyTo foreach { _ ! WorkflowManagerSubmitFailure(e)}
-        // Return the original state data if the workflow failed submission.
-        stateData
-      case Success(a) =>
-        a.actor ! (if (isRestart) Restart else Start(replyTo))
-        a.data
+    implicit val actorSystem = context.system
+    val materializedWfDescriptorFuture = PromiseActor.askNoTimeout(MaterializeWorkflowDescriptorActor.MaterializeWorkflow(workflowId, source, config),
+      CromwellServer.workflowDescriptorMaterializeActor).mapTo[MaterializationResult]
```

Comment:
Alternatively, as actors create other actors, the `WorkflowManagerActor` could conceivably create its own `workflowDescriptorMaterializeActor` running in the same system.